### DISCUSSION TITLE
How To Jump To Bottom Of Scroll Area

### DISCUSSION THREAD

**Discussion by Existential-Critic** [shadcn-ui/ui#2901] [Q&A]
I am trying to use a Scroll Area as a chat log, and I want to make a function so when a new message is added to the log, it jumps the Scroll Area to the bottom. Is there a function to do that, and if not, how would I go about setting one up?

**Comment by pablotanner** [comment]
If the scroll area has a scroll bar, you can use the following guide

https://stackoverflow.com/questions/270612/scroll-to-bottom-of-div

**Reply by Existential-Critic** [reply]
I will check it out, thank you!

**Reply by vishalvibes** [reply]
@Existential-Critic Were you able to solve this?

**Reply by Existential-Critic** [reply]
I was not unfortunately, I had to do it using refs and removing the ScrollArea.

**Reply by vishalvibes** [reply]
I did find a solution:

1️⃣ I added a ref to the div which was the children of ScrollBar instead of ScrollBar and it worked

<img width="587" alt="Screenshot 2024-04-02 at 9 02 30 AM" src="https://github.com/shadcn-ui/ui/assets/29145212/bbc3ab75-361e-44a5-b792-fa044c8ee802">

2️⃣ Then I used the scrollIntoView(false) function

<img width="454" alt="Screenshot 2024-04-02 at 9 04 24 AM" src="https://github.com/shadcn-ui/ui/assets/29145212/76fa9293-3bd8-42c0-83d1-737d49ecb6d2">

Hope this helps.

**Reply by kamiljaneczek** [reply]
works like charm. thank you :)

**Reply by dankags** [reply]
it is a really good article and it works fine the issue on my code was i did not know that you should put scrollintoview boolean parameter to false since by default is true and it keeps it at the top.

**Reply by MijailVillegas** [reply]
> I did find a solution:
> 
> 1️⃣ I added a ref to the div which was the children of ScrollBar instead of ScrollBar and it worked
> 
> <img alt="Screenshot 2024-04-02 at 9 02 30 AM" width="587" src="https://private-user-images.githubusercontent.com/29145212/318630987-bbc3ab75-361e-44a5-b792-fa044c8ee802.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MjcxOTUsIm5iZiI6MTcyNTcyNjg5NSwicGF0aCI6Ii8yOTE0NTIxMi8zMTg2MzA5ODctYmJjM2FiNzUtMzYxZS00NGE1LWI3OTItZmEwNDRjOGVlODAyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA3VDE2MzQ1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTlmZTQyMTA5MDMxNDBlZmQ2ODQ4MTE4ZjVjZGE0YTE0MmFmZmI3NjE4YTQwNTQ5NWRmNDU2MmYzODFmMzU5NGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.u3MYyRuJGz95u1lpTjZKR3vveQlTXCIyCmMn-TVqP4I">
> 2️⃣ Then I used the scrollIntoView(false) function
> 
> <img alt="Screenshot 2024-04-02 at 9 04 24 AM" width="454" src="https://private-user-images.githubusercontent.com/29145212/318631072-76fa9293-3bd8-42c0-83d1-737d49ecb6d2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU3MjcxOTUsIm5iZiI6MTcyNTcyNjg5NSwicGF0aCI6Ii8yOTE0NTIxMi8zMTg2MzEwNzItNzZmYTkyOTMtM2JkOC00MmMwLTgzZDEtNzM3ZDQ5ZWNiNmQyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA3VDE2MzQ1NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNhNjcyMzIwMDQzYzg0MmIyNDg1MjI5ZDI0NjgxM2I0YzJkNThjYzVjYWUzNzgyYjY5YWJjOTVhMWFhMzIzZTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.m9hhcNml9LQ17aixmLFj1KboWp6SSHNMJybtWb6MyEs">
> Hope this helps.

Thank you so much

**Reply by CBell045** [reply]
You can also extend the scroll area as described here: https://github.com/radix-ui/primitives/discussions/990#discussioncomment-9335976

**Comment by edlucas** [comment]
I have a version where I scroll the last chat message into view using this utility function that takes the wrapper as a parameter:

```
export const scrollToBottom = (
  container: HTMLElement | null,
  smooth = false,
) => {
  if (container?.children.length) {
    const lastElement = container?.lastChild as HTMLElement

    lastElement?.scrollIntoView({
      behavior: smooth ? 'smooth' : 'auto',
      block: 'end',
      inline: 'nearest',
    })
  }
}
```

As in the previous example, I'm getting using a `div` within `<ScrollArea>`, giving it a `ref` and calling the function in a `useEffect()`.

**Comment by szmazhr** [comment]
Don't know if a over engineered or not. but here is my solution. Which keeps it stick to bottom even when new element added.

https://blog.shahzar.in/how-to-add-stick-to-bottom-features-to-shadcn-scrollarea-for-better-chat-interfaces
